src/pipeline_migration/actions/add_task.py

```diff
@@ -0,0 +1,406 @@
+        "e.g. registry/org/task-name:tag@digest."
+        "If omitted, the latest task bundle is queried from the registry.",
+    )
+    add_task_parser.add_argument(
+        "-g",
+        "--git-add",
+        dest="git_add",
+        action="store_true",
+        help="Add the modified files to git index.",
+    )
+    add_task_parser.set_defaults(action=action)
+
+
+def action(args) -> None:
+    actual_task_name: Final = args.task
+    pipeline_task_name: Final = args.pipeline_task_name or args.task
+
+    bundle_ref = args.bundle_ref
+    if not bundle_ref:
+        bundle_ref = KonfluxBuildDefinitions().query_latest_bundle(actual_task_name)
+
+    logger.info("Adding task %s, bundle %s", actual_task_name, bundle_ref)
+
+    task_config = {
+        "name": pipeline_task_name,
+        "taskRef": {
+            "resolver": "bundles",
+            "params": [
+                {"name": "kind", "value": "task"},
+                {"name": "name", "value": actual_task_name},
+                {"name": "bundle", "value": bundle_ref},
+            ],
+        },
+    }
+    if args.params:
+        task_config["params"] = [{"name": name, "value": value} for name, value in args.params]
+    if args.run_after:
+        task_config["runAfter"] = args.run_after
+    if args.skip_checks:
+        task_config["when"] = [
+            {"input": "$(params.skip-checks)", "operator": "in", "values": [False]}
+        ]
+
+    search_places = [path for path in args.file_or_dir if path]
+    relative_tekton_dir = Path("./.tekton")
+    if not search_places and relative_tekton_dir.exists():
+        search_places = [str(relative_tekton_dir.absolute())]
+
+    for original_abs_path, pipeline_file in search_pipeline_files(search_places):
+        style = YAMLStyle.detect(pipeline_file)
+        doc = load_yaml(pipeline_file)
+        tasks = doc["spec"]["tasks"]
+
+        existing_pipeline_task_names = set([])
+        existing_actual_task_names = set([])
+
+        for name1, name2 in KonfluxBuildDefinitions.extract_task_names(tasks):
+            existing_pipeline_task_names.add(name1)
+            existing_actual_task_names.add(name2)
+
+        if (depended_tasks := task_config.get("runAfter")) is not None:
+            for name in depended_tasks:
+                if name not in existing_pipeline_task_names:
+                    raise ValueError(
+                        f"Task {name} does not exist in the pipeline definition {pipeline_file}."
+                    )
+
+        if pipeline_task_name in existing_pipeline_task_names:
+            logger.info(
+                "Task %s is included in pipeline %s already.", pipeline_task_name, pipeline_file
+            )
+            continue
+
+        if actual_task_name in existing_actual_task_names:
+            logger.info(
+                "Task %s is being referenced in pipeline %s already.",
+                actual_task_name,
+                pipeline_file,
+            )
+            continue
```

Comment:
I don't think we need to skip in this case. There can be legitimate reasons why a user would want to run the same task twice with different pipeline-task-name